tests/devices/qubit/test_preprocess.py

```diff
@@ -382,19 +396,22 @@ def test_not_expval(self):
         )
 
     def test_unsupported_op(self):
-        """Test if a QuantumFunctionError is raised for an unsupported operation, i.e.,
-        multi-parameter operations that are not qml.Rot"""
+        """Test that a device supported by the forward pass but"""
```

Comment:
name and docstring need updating. also, why did U2 gain support, deeper decomposition?

Reply:
This PR also updates the preprocessing to perform deeper decompositions  (till no variables or one variable with a generator) in the situation where adjoint differentiation was requested.